### ISSUE TITLE
Dialog: Close Buttons must have discernible text

### ISSUE THREAD

**Issue by HellinSunWTW** [primefaces/primeng#14977]
### Describe the bug

The Axe Dev Tools for Accessibility is reporting that the 'Buttons must have discernible text':
![image](https://github.com/primefaces/primeng/assets/152492138/c75841fd-0d6c-4cba-9420-3127f761e0c9)
Can we have appropriate aria label like "Close the dialog" added please?

### Environment

Windows 10

### Angular version

17.2.3

### PrimeNG version

17.9.0

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

18.18.0

### Steps to reproduce the behavior

1. Install the Axe Dev Tools for Accessibility plug-in.
2. Visit the PrimeNG dialog page (https://primeng.org/dialog) on Chrome/Edge browser.
3. Scan the page with Axe

**Comment by TaneliTuomola** [comment]
There is actually a undocumented property "closeAriaLabel" that can be used to set aria-label to the close button.
`<p-dialog header="Header" closeAriaLabel="Close">`
Documents should be updated.

**Comment by Draganlazarev90** [comment]
> There is actually a undocumented property "closeAriaLabel" that can be used to set aria-label to the close button. `<p-dialog header="Header" closeAriaLabel="Close">` Documents should be updated.

this doesn't work with dynamic dialog. When you add the closeAriaLabel in the config it still doesn't apply it to the close button

**Comment by TaneliTuomola** [comment]
> > There is actually a undocumented property "closeAriaLabel" that can be used to set aria-label to the close button. `<p-dialog header="Header" closeAriaLabel="Close">` Documents should be updated.
> 
> this doesn't work with dynamic dialog. When you add the closeAriaLabel in the config it still doesn't apply it to the close button

https://github.com/primefaces/primeng/issues/15144

**Comment by itworksafisher** [comment]
Shouldn't this be using primeng [setTranslations](https://primeng.org/configuration#locale)? I would expect that to have a close property that would then be used by default for the dialogs rather than requiring consumer to provide one to each instance.
